static/js/user_profile.js

```diff
@@ -47,10 +53,11 @@ function render_user_stream_list(streams, user) {
     streams.sort(compare_by_name);
     const container = $("#user-profile-modal .user-stream-list");
     container.empty();
+    const show_unsubscribe_button = people.is_my_user_id(user.user_id) || page_params.is_admin;
```

Comment:
Can we make the `is_admin` part of this conditional call a new `user_can_subscribe_other_users` function in `settings_data.js` next to `user_can_subscribe_other_users`?  I know it's kinda trivial, but I think it'd help with the readability of the codebase to use that pattern.

Can be a small prep commit.